src/Hooks/useLoadMovies.ts

```diff
@@ -1,32 +1,21 @@
 import { useEffect } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
-import StateType from '../Types/StateType';
+import RootState from '../Types/RootState';
+import { handleLoadingAction } from '../state/reducerActions';
+import { useAppDispatch } from '../state/store';
+import fetchMovies from '../asyncActions/fetchMovies';
 
 export default function useLoadMovies() {
   const dispatch = useDispatch();
+  const appDispatch = useAppDispatch();
 
   const { movies, isLoading, error, sort, filter, totalAmount, offset } =
-    useSelector((state: StateType) => state);
+    useSelector((state: RootState) => state.movieState);
 
-  // TODO: thunk
   useEffect(() => {
-    dispatch({ type: 'handleLoading' });
-    fetch(
-      `http://localhost:4000/movies?sortBy=${sort}&sortOrder=desc${
-        filter !== 'all' ? `&filter=${filter}` : ''
-      }&offset=${offset}&limit=9`
-    )
-      .then((response) => response.json())
-      .then((data) => {
-        dispatch({
-          type: 'handleMovies',
-          payload: { movies: data.data, totalAmount: data.totalAmount },
-        });
-      })
-      .catch((catchedError) => {
-        dispatch({ type: 'handleError', payload: catchedError });
-      });
-  }, [dispatch, sort, filter, offset]);
+    dispatch(handleLoadingAction());
+    appDispatch(fetchMovies(filter, sort, offset));
```

Comment:
Let's move dispatch(handleLoadingAction()); to fetchMovies.

Reply:
Moved this dispatch to fetchMovies.